src/js/game/LevelMVC/FacingDirection.js

```diff
@@ -23,7 +40,7 @@ const FacingDirection = Object.freeze({
   },
 
   turn: function (facing, rotation) {
-    return (facing + 4 + (rotation === 'right' ? 1 : -1)) % 4;
+    return (facing + 4 + (rotation === 'right' ? 1 : -1)) % 4; // still speechless that JS has a mod bug...
```

Comment:
haha, didn't mean to leave that in but:
The ... + 4 .... % 4 baffled me, because that literally should do nothing (as 1 % 4 === 5 % 4)
BUT
it's to keep the a value (of a%b) in the positive range, because js returns incorrect mod values when a < 0.

-1 % 4 = 3
(because a%b = [a/b - floor(a/b)] * b
or in this case: -1 % 4 = [-1/4 - floor(-1/4)] * 4
                                       [-1/4 - (-1)] * 4
                                        [3/4] * 4
                                              3

yet in js -1 % 4 returns -1, and I would bet MONEY that it's because the operator has a 
if (a < b) {return a;} check, which is appropriate with positive numbers, but not negative numbers.

😂 

just chuckles.

Reply:
Oh, it's actually because `%` in JS is [the _remainder_ operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Arithmetic_Operators#Remainder) not actually modulo